frontend/src/app/shared/dark-mode-toggler/dark-mode-toggler.component.ts

```diff
@@ -0,0 +1,29 @@
+import { ChangeDetectionStrategy, Component, HostBinding } from '@angular/core';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { DarkModeTogglerService } from '../../core/service/dark-mode.service';
+
+@Component({
+  selector: 'app-dark-mode-toggler',
+  standalone: true,
+  imports: [MatButtonModule, MatIconModule],
+  templateUrl: './dark-mode-toggler.component.html',
+  styleUrl: './dark-mode-toggler.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
+export class DarkModeTogglerComponent {
+  @HostBinding('class.dark-mode') isDarkMode = false;
```

Comment:
we may need to set the class on the body, to have effect on the whole page, like here https://dev.to/omamaaslam/a-beginners-guide-to-adding-lightdark-mode-toggle-in-angular-material-47h7